src/main/java/com/google/edith/GroceryDataReader.java

```diff
@@ -0,0 +1,105 @@
+package com.google.edith;
+
+import com.opencsv.CSVReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+/** 
+  * Processes data file of product prices and sizes at different
+  * stores and returns the best value item. 
+  */
+public final class GroceryDataReader {
+
+    private static final String ALDI = "Aldi";
+    private static final String KROGER = "Kroger";
+    private static final String TRADERJOES = "Trader Joe's";
+    private static final String PUBLIX = "Publix";
+    private static final String WALMART = "Walmart";
+    
+  /**
+    * Finds the specified product in the file and puts the
+    * data into DealItem objects to be handled. 
+    */
+  public DealItem readFile(String itemName) throws IOException {
+    URL csvResource = getClass().getClassLoader().getResource("grocerydata.csv");
+    File groceryDataFile = new File(csvResource.getFile());
+
+    CSVReader reader = new CSVReader(new FileReader(groceryDataFile), ',');
+    DealItem cheapestItem = null;
+
+    String[] record = null;
+    record = reader.readNext();
+    record = reader.readNext();
+
+	while ((record = reader.readNext()) != null) {
+      if (record[0].equals(itemName)) {
+        List<DealItem> dealItems = new ArrayList<DealItem>();
+
+        DealItem item1 = new DealItem();
+        item1.setStore(ALDI);
+        item1.setPrice(record[1]);
+        item1.setWeight(record[2]);
+        item1.setComment(record[3]);
+        dealItems.add(item1);
```

Comment:
I think if the order of the stores is always the same maybe we could modify lines 42-75 into a loop to avoid duplicate logic
```
private static final ImmutableList<String> STORES = ImmutableList.of(KROGER, TRADERJOES, ..);
for (int i = 0; i < STORES.size(); i++) {
  DealItem item = new DealItem();
  item.setStore(STOERES.get(i));
  item.setPrice(record[i+1]);
  item.setWeight(record[i+2]);
  item.setComment(record[i+3]);
  dealItems.add(item);
}
cheapestItem = getCheapest..();
```
I might have missed something so this may not work, but worth considering if it does.

Reply:
it works! i had to use record[i*3+1] to get the proper indices